File: Recursion/factorial.py
# basic factorial problem using recursion:

def factorial(number):
  if number == 0:
    return 1
  elif number < 0:
    return -1
  return number * factorial(number - 1)

# I dont know the use of this name == main, gonna update after research:
if __name__ == '__main__':
  # if user enters something other than an integer value:
  try:
    number = int(input("Please enter a number: "))
  except:
    number = int(input("Input Error! Please enter the number again: "))
  finally:
    if factorial(number) < 0:
      number = int(input("Please enter a whole number: "))
    print("Factorial of this number is: " + str(factorial(number)))

File: CPP/Recursion/factorial.cpp
#include <iostream>
using namespace std;

int factorial(int number){
  if (number <= 0)
    return 1;
  
  return (number*factorial(number-1));
}

int main(){
  int number = 0;
  cout << "Please enter a number: "; 
  cin >> number;
  if (number < 0){
    cout << endl << "Please enter a whole number: ";
    cin >> number;
  }
  cout << endl << "Factorial of that number is: " << factorial(number) << endl;
  return 0; 
}